public/app/plugins/gen.go

```diff
@@ -18,15 +19,28 @@ import (
 
 	corecodegen "github.com/grafana/grafana/pkg/codegen"
 	"github.com/grafana/grafana/pkg/cuectx"
+	"github.com/grafana/grafana/pkg/kinds"
+	"github.com/grafana/grafana/pkg/plugins"
 	"github.com/grafana/grafana/pkg/plugins/codegen"
-	"github.com/grafana/grafana/pkg/plugins/pfs"
-	"github.com/grafana/thema"
+	"github.com/grafana/grafana/pkg/plugins/config"
+	"github.com/grafana/grafana/pkg/plugins/manager/fakes"
+	"github.com/grafana/grafana/pkg/plugins/manager/loader"
+	"github.com/grafana/grafana/pkg/plugins/manager/loader/angular/angularinspector"
+	"github.com/grafana/grafana/pkg/plugins/manager/loader/assetpath"
+	"github.com/grafana/grafana/pkg/plugins/manager/loader/finder"
+	"github.com/grafana/grafana/pkg/plugins/manager/registry"
+	"github.com/grafana/grafana/pkg/plugins/manager/signature"
+	"github.com/grafana/grafana/pkg/plugins/manager/signature/statickey"
+	"github.com/grafana/grafana/pkg/plugins/manager/sources"
+	"github.com/grafana/grafana/pkg/plugins/manager/store"
+	"github.com/grafana/grafana/pkg/plugins/pluginscdn"
+	"github.com/grafana/grafana/pkg/setting"
```

Comment:
We've generally found that keeping the codegen path separate from other runtime backend code is pretty important for maintainability. The simplest issue we ran into before was that allowing imports into the main codebase created loops in development on jennies where it became possible for `make gen-cue` to break because of the faulty output of a prior `make gen-cue` run. (i've spent hours in those loops, they're really confusing to debug)

Also, having all these imports (and the things they reference in the jenny signatures) will make it impossible for us to move/import these into kindsys, which we want to do in order to create a robust codegen test regression suite in a single, central place.

Is it possible to keep it to a minimal set of imports that avoid pulling in most of the backend? IMO it's actually fine and worth writing a bit of duplicate code here in order to avoid the possibility of the loops described above.

Reply:
That make sense given our similar approach in pkg/plugins (plugin management) don't import dependencies from the outside.